src/Symfony/Component/Serializer/Serializer.php

```diff
@@ -177,31 +177,58 @@ public function denormalize($data, $type, $format = null, array $context = array
     /**
      * {@inheritdoc}
      */
-    public function supportsNormalization($data, $format = null)
+    public function supportsNormalization($data, $format = null/*, array $context = array()*/)
     {
-        return null !== $this->getNormalizer($data, $format);
+        if (func_num_args() > 2) {
+            $context = func_get_arg(2);
+        } else {
+            if (__CLASS__ !== get_class($this)) {
+                $r = new \ReflectionMethod($this, __FUNCTION__);
+                if (__CLASS__ !== $r->getDeclaringClass()->getName()) {
+                    @trigger_error(sprintf('Method %s() will have a third `$context = array()` argument in version 4.0. Not defining it is deprecated since 3.3.', get_class($this), __FUNCTION__), E_USER_DEPRECATED);
```

Comment:
Overriding this class is common (even if it's not a good practice IMO). Not sure it is worth it to make it final.